### PR TITLE
quote `import_path` when importing in nix

### LINKED ISSUE #190: `nix-update` crashes on paths that contain spaces
If your current working directory’s path contains spaces, then `nix-update` will crash:

```
bash-5.2$ nix-update hello
$ nix eval --json --impure --expr 
let
  
  pkgs = import /home/jayman/Path that contains spaces/nixpkgs;
  args =  builtins.functionArgs pkgs;
  inputs = (if args ? system then { system = builtins.currentSystem; } else {}) //
           (if args ? overlays then { overlays = [ ]; } else {});
  pkg = (pkgs inputs)."hello";
  sanitizePosition = x: x;

  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);

  position = if pkg ? isRubyGem then
    raw_version_position
  else if pkg ? isPhpExtension then
    raw_version_position
   else
    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);
in {
  name = pkg.name;
  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;
  inherit raw_version_position;
  filename = position.file;
  line = position.line;
  urls = pkg.src.urls or null;
  url = pkg.src.url or null;
  rev = pkg.src.rev or null;
  hash = pkg.src.outputHash or null;
  go_modules = pkg.goModules.outputHash or null;
  go_modules_old = pkg.go-modules.outputHash or null;
  cargo_deps = pkg.cargoDeps.outputHash or null;
  raw_cargo_lock =
    if pkg ? cargoDeps.lockFile then
      let
        inherit (pkg.cargoDeps) lockFile;
        res = builtins.tryEval (sanitizePosition {
          file = toString lockFile;
        });
      in
      if res.success then res.value.file else false
    else
      null;
  npm_deps = pkg.npmDeps.outputHash or null;
  tests = builtins.attrNames (pkg.passthru.tests or {});
  has_update_script = pkg.passthru.updateScript or null != null;
  src_homepage = pkg.src.meta.homepage or null;
  changelog = pkg.meta.changelog or null;
} --extra-experimental-features flakes nix-command
error: undefined variable 'that'

       at «string»:4:35:

            3|
            4|   pkgs = import /home/jayman/Path that contains spaces/nixpkgs;
             |                                   ^
            5|   args =  builtins.functionArgs pkgs;
Traceback (most recent call last):
  File "/nix/store/yp0i8s3hq90h5kqsxq0hlgyxrhswpqjn-nix-update/bin/.nix-update-wrapped", line 9, in <module>
    sys.exit(main())
             ^^^^^^
  File "/nix/store/yp0i8s3hq90h5kqsxq0hlgyxrhswpqjn-nix-update/lib/python3.11/site-packages/nix_update/__init__.py", line 287, in main
    package = update(options)
              ^^^^^^^^^^^^^^^
  File "/nix/store/yp0i8s3hq90h5kqsxq0hlgyxrhswpqjn-nix-update/lib/python3.11/site-packages/nix_update/update.py", line 318, in update
    package = eval_attr(opts)
              ^^^^^^^^^^^^^^^
  File "/nix/store/yp0i8s3hq90h5kqsxq0hlgyxrhswpqjn-nix-update/lib/python3.11/site-packages/nix_update/eval.py", line 179, in eval_attr
    res = run(cmd)
          ^^^^^^^^
  File "/nix/store/yp0i8s3hq90h5kqsxq0hlgyxrhswpqjn-nix-update/lib/python3.11/site-packages/nix_update/utils.py", line 36, in run
    return subprocess.run(
           ^^^^^^^^^^^^^^^
  File "/nix/store/px3zbr30lbnmb3drlx5djwdd792l9vwz-python3-3.11.4/lib/python3.11/subprocess.py", line 571, in run
    raise CalledProcessError(retcode, process.args,
subprocess.CalledProcessError: Command '['nix', 'eval', '--json', '--impure', '--expr', '\nlet\n  \n  pkgs = import /home/jayman/Path that contains spaces/nixpkgs;\n  args =  builtins.functionArgs pkgs;\n  inputs = (if args ? system then { system = builtins.currentSystem; } else {}) //\n           (if args ? overlays then { overlays = [ ]; } else {});\n  pkg = (pkgs inputs)."hello";\n  sanitizePosition = x: x;\n\n  raw_version_position = sanitizePosition (builtins.unsafeGetAttrPos "version" pkg);\n\n  position = if pkg ? isRubyGem then\n    raw_version_position\n  else if pkg ? isPhpExtension then\n    raw_version_position\n   else\n    sanitizePosition (builtins.unsafeGetAttrPos "src" pkg);\nin {\n  name = pkg.name;\n  old_version = pkg.version or (builtins.parseDrvName pkg.name).version;\n  inherit raw_version_position;\n  filename = position.file;\n  line = position.line;\n  urls = pkg.src.urls or null;\n  url = pkg.src.url or null;\n  rev = pkg.src.rev or null;\n  hash = pkg.src.outputHash or null;\n  go_modules = pkg.goModules.outputHash or null;\n  go_modules_old = pkg.go-modules.outputHash or null;\n  cargo_deps = pkg.cargoDeps.outputHash or null;\n  raw_cargo_lock =\n    if pkg ? cargoDeps.lockFile then\n      let\n        inherit (pkg.cargoDeps) lockFile;\n        res = builtins.tryEval (sanitizePosition {\n          file = toString lockFile;\n        });\n      in\n      if res.success then res.value.file else false\n    else\n      null;\n  npm_deps = pkg.npmDeps.outputHash or null;\n  tests = builtins.attrNames (pkg.passthru.tests or {});\n  has_update_script = pkg.passthru.updateScript or null != null;\n  src_homepage = pkg.src.meta.homepage or null;\n  changelog = pkg.meta.changelog or null;\n}', '--extra-experimental-features', 'flakes nix-command']' returned non-zero exit status 1.
bash-5.2$ 
```

### PR DESCRIPTION
**Pull request by figsoda**
fixes #190

### CODE REVIEW

File: nix_update/update.py [hunk 1]
```diff
@@ -86,7 +86,7 @@ def get_package(opts: Options) -> str:
     return (
         f'(let flake = builtins.getFlake "{opts.import_path}"; in flake.packages.${{builtins.currentSystem}}.{opts.escaped_attribute} or flake.{opts.escaped_attribute})'
         if opts.flake
-        else f"(import {opts.import_path} {disable_check_meta(opts)}).{opts.escaped_attribute}"
+        else f'(import "{opts.import_path}" {disable_check_meta(opts)}).{opts.escaped_attribute}'
```
Comment (Mic92) [inline nix_update/update.py:89]: I think we would need to make this an absolute path first.

Comment (Mic92) [reply nix_update/update.py:89]: Ah. It's already using realpath.

File: nix_update/__init__.py [hunk 2]
```diff
@@ -125,7 +125,7 @@ def nix_shell(options: Options) -> None:
             check=False,
         )
     else:
-        expr = f"let pkgs = import {options.import_path} {{}}; in pkgs.mkShell {{ buildInputs = [ pkgs.{options.escaped_attribute} ]; }}"
+        expr = f'let pkgs = import "{options.import_path}" {{}}; in pkgs.mkShell {{ buildInputs = [ pkgs.{options.escaped_attribute} ]; }}'
```
Comment (Mic92) [inline nix_update/__init__.py:128]: Should we also use json.dumps on import_path to escape characters like `"`?

Comment (figsoda) [reply nix_update/__init__.py:128]: yeah that's probably better

**Comment by Jayman2000** [comment]
I just tested this, and it seems to work for both paths that contain spaces and paths that contain quotation marks.

**Pull request merged**